### PR TITLE
Stabilize namings cardano transactions

### LINKED ISSUE #1535: Stabilize namings in `Cardano Transactions` API
## Why
We want to stabilize the naming of the new Cardano transactions API.

## What
Rename the `commitment` sub command of the Mithril client CLI with `snapshot`, and rename the command `snapshot` with `cardano-db` (while keeping `snapshot` as an alias to avoid breaking change now). Harmonize the naming of the `Commitment` related types and replace with `Snapshot`.

## How

- [x] Rename command/sub-command of mithril client CLI (and warn user when using deprecated alias?)
- [x] Rename `Commitment`  related types in all the repository
- [x] Update documentation

### PR DESCRIPTION
**Pull request by dlachaumepalo**
## Content
This PR includes a refactor after the recent implementation of Cardano transactions.

- Cardano transaction **_commitment_** has been renamed Cardano transaction **_snapshot_** globally in the repository.
- Mithril client CLI command `snapshot` has been renamed `cardano-db`. The `snapshot` command still exists to avoid breaking changes but is deprecated.

## Pre-submit checklist

- Branch
  - [ ] Tests are provided (if possible)
  - [x] Crates versions are updated (if relevant)
  - [x] Commit sequence broadly makes sense
  - [x] Key commits have useful messages
- PR
  - [x] No clippy warnings in the CI
  - [x] Self-reviewed the diff
  - [x] Useful pull request description
  - [x] Reviewer requested
- Documentation
  - [x] Update README file (if relevant)
  - [x] Update documentation website (if relevant)
  - [ ] Add dev blog post (if relevant)

## Issue(s)
Closes #1535

### CODE REVIEW

**Review by Alenar — APPROVED** [review]

**Review by sfauvel — APPROVED** [review]

**Review by jpraynaud — COMMENTED** [review]
Looks good overall. As seen together, we should probably have the following commands/sub-commands:
- `snapshot list` => `cardano-db snapshot list`
- `snapshot show` => `cardano-db snapshot show`
- `snapshot download` => `cardano-db download` or `cardano-db restore`

File: mithril-client-wasm/src/client_wasm.rs
[the remark this answers is not in the document]
Comment (Alenar) [reply mithril-client-wasm/src/client_wasm.rs:1]: I think we don't, we should bump it (and start to do so for each breaking changes on the transactions) after we stabilize the api.

File: docs/website/root/manual/developer-docs/nodes/mithril-client.md [hunk 1]
```diff
@@ -139,13 +140,13 @@ Options:
       --config-directory <CONFIG_DIRECTORY>
           Directory where configuration file is located [default: ./config]
       --aggregator-endpoint <AGGREGATOR_ENDPOINT>
-          Override configuration Aggregator endpoint URL [env: AGGREGATOR_ENDPOINT=]
+          Override configuration Aggregator endpoint URL [env: AGGREGATOR_ENDPOINT=https://aggregator.testing-sanchonet.api.mithril.network/aggregator]
```
[the remark this answers is not in the document]
Comment (Alenar) [reply docs/website/root/manual/developer-docs/nodes/mithril-client.md:143]: I will remove it from this doc.
It's just that when we run a `mithril-client --help` to copy it's output to the docs there was the sanchonet aggregator set in the `AGGREGATOR_ENDPOINT` env var.

File: mithril-client-cli/src/main.rs [hunk 2]
```diff
@@ -165,7 +165,7 @@ impl Source for Args {
 enum ArtifactCommands {
     /// Deprecated, use `cardano-db` instead
     #[clap(subcommand)]
-    #[deprecated(since = "0.7.2", note = "use `CardanoDb` commands instead")]
+    #[deprecated(since = "0.7.3", note = "use `CardanoDb` commands instead")]
```
Comment (jpraynaud) [inline mithril-client-cli/src/main.rs:168]:
```suggestion
    #[deprecated(since = "0.7.3", note = "use `cardano-db` commands instead")]
```

**Review by jpraynaud — APPROVED** [review]
LGTM :+1:

File: mithril-client-cli/src/main.rs (hunk 2 above)
Comment (Alenar) [reply mithril-client-cli/src/main.rs:168]: This is only in the code is not shown to users, so for me it make more sense to use the enum variant name than the its resulting command line value _(honestly this line is quite superfluous, we put in order to test how rust handles depreciation)_.

**Pull request merged**